runtime/parachains/src/ump/benchmarking.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::{Pallet as Ump, *};
+use frame_system::RawOrigin;
+use xcm::prelude::*;
+
+fn assert_last_event<T: Config>(generic_event: <T as Config>::Event) {
+	let events = frame_system::Pallet::<T>::events();
+	let system_event: <T as frame_system::Config>::Event = generic_event.into();
+	// compare to the last event record
+	let frame_system::EventRecord { event, .. } = &events[events.len() - 1];
+	assert_eq!(event, &system_event);
+}
+
+fn assert_last_event_type<T: Config>(generic_event: <T as Config>::Event) {
+	let events = frame_system::Pallet::<T>::events();
+	let system_event: <T as frame_system::Config>::Event = generic_event.into();
+	// compare to the last event record
+	let frame_system::EventRecord { event, .. } = &events[events.len() - 1];
+	assert_eq!(sp_std::mem::discriminant(event), sp_std::mem::discriminant(&system_event));
+}
+
+fn queue_upward_msg<T: Config>(
+	host_conf: &HostConfiguration<T::BlockNumber>,
+	para: ParaId,
+	msg: UpwardMessage,
+) {
+	let len = msg.len() as u32;
+	let msgs = vec![msg];
+	Ump::<T>::check_upward_messages(host_conf, para, &msgs).unwrap();
+	let _ = Ump::<T>::receive_upward_messages(para, msgs);
+	assert_last_event::<T>(Event::UpwardMessagesReceived(para, 1, len).into());
+}
+
+fn create_message<T: Config>(weight: u64, size: u32) -> Vec<u8> {
+	// fill the remark but leave some bytes for the encoding of the message
+	let size = size.saturating_sub(20) as usize;
+	let mut remark = Vec::new();
+	remark.resize(size, 0u8);
+	let call = frame_system::Call::<T>::remark_with_event { remark };
+	VersionedXcm::<T>::from(Xcm::<T>(vec![Transact {
+		origin_type: OriginKind::SovereignAccount,
+		require_weight_at_most: weight,
+		call: call.encode().into(),
+	}]))
+	.encode()
+}
+
+frame_benchmarking::benchmarks! {
+	process_upward_message {
+		let s in 0..configuration::ActiveConfig::<T>::get().max_upward_message_size;
```

Comment:
this is basically using the value of the `ActiveConfig` in genesis, not the real on on-chain.

I feel like coding it like this is actually misleading. A better approach would be to hardcode some maximum value for it, and ensuring that when setting it, we don't go over it. See an example here: https://github.com/paritytech/polkadot/pull/3876/files

Reply:
I see, but how can I determine this upper bound? 
Isn't it a delicate point bounding on-chain values, which in principle stand under the control of democracy.

Reply:
Some reasonable defaults should be enough, and no if some upper bounds exists even the governance should not (easily) be able to set higher values. @pepyakin might be able to chime in here.

Reply:
I meant it the other way around. Governance  should not be **limited** by bounds set in code by a single developer.

But I'm still a noob and have not so much insight in the delicate gears that make a chain safe and governance truly  democratic. It was just a feeling and an open question.

Reply:
> I meant it the other way around. Governance should not be limited by bounds set in code by a single developer.

The governance is not and will not be bound at the end of the day by any of these because by the same token that they can set this parameter, they can also change the code that contains this parameter.

I agree that the value should not be arbitrary, therefore I mentioned someone more knowledgable about parachains.

Reply:
This is not fixed.

Reply:
I need a hint for sensible bound values.

Reply:
@kianenigma @pepyakin Can someone give me a hint for reasonable default values? I really like to get this PR landed before christmas 😄

Reply:
Sorry.

Currently on kusama and polkadot this value is set to 50 KiB. I think this is a sensible value. I think it may be possible we decrease this value even. (Decreasing would mean that the value would go out of sync with the benchmarking but that's OK as long as it goes into the conservative direction)

Reply:
@kianenigma I added a max-bound as suggested with the value @pepyakin proposed. Can I have another review please?